src/candid_rpc.rs

```diff
@@ -61,13 +58,23 @@ impl RpcTransport for CanisterTransport {
     async fn http_request(
         _provider: &RpcService,
         request: CanisterHttpRequestArgument,
-        cost: u128,
-    ) -> CallResult<HttpResponse> {
-        Ok(
-            ic_cdk::api::management_canister::http_request::http_request(request, cost)
-                .await?
-                .0,
-        )
+        cycles: u128,
+    ) -> RpcResult<HttpResponse> {
+        if !is_authorized(&ic_cdk::caller(), Auth::FreeRpc) {
```

Comment:
The cycles payment logic was originally in the forked ckETH codebase; I moved it here to support the `FreeRpc` authorization for Candid-RPC methods. Everything else works the same as before.